### PR TITLE
Added support for a config file

### LINKED ISSUE #118: Use config file to ignore specific rules
Would be nice to have config file like `.hadolintrc` to disable specific rules check
Most linters have this feature

### PR DESCRIPTION
**Pull request by lorenzo**
CLI options take precedence over the config file
Closes #118

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -38,19 +53,53 @@ printChecks checks = do
...
+applyConfig o@(LintOptions _ (_:_) _) = return o
+applyConfig o = do
+    theConfig <- findConfig
+    case theConfig of
+        Nothing -> return o
+        Just config -> parseAndApply config
+  where
+    findConfig = do
+        localConfigFile <- (</> "hadolint.yaml") <$> getCurrentDirectory
+        configFile <- getXdgDirectory XdgConfig "hadolint.yaml"
```
Comment (zemanlx) [inline app/Main.hs]: Would you mind to have it hidden by defaul, eg `.hadolint.yaml`?

**Comment by zemanlx** [comment]
I have also asked appveyor to increase a limit for our project so I hope it stops failing soon.

File: app/Main.hs (hunk 1 above)
Comment (lorenzo) [reply app/Main.hs]: Support both hidden and not hidden?

Comment (zemanlx) [reply app/Main.hs]: I think hidden(starting with `.`) like `.travis.yml` or `.hindent.yaml` or `.zshrc` is enough.

**Comment by zemanlx** [comment]
When you will be editing README, can you please also change https://github.com/hadolint/hadolint#ast appropriately, when `Syntax.hs` is no longer part of hadolint?

**Comment by lorenzo** [comment]
@zemanlx done

**Comment by lorenzo** [comment]
Only the local one is take in account if both exist

**Comment by lorenzo** [comment]
@zemanlx any answers from appveyor?

**Comment by zemanlx** [comment]
I have just got a response a few minutes ago...rebuilding now. I have noticed that binaries are now much bigger. 

[16MB](https://ci.appveyor.com/project/hadolint/hadolint/build/1.0.75/artifacts) vs [26MB](https://ci.appveyor.com/project/hadolint/hadolint/build/1.0.81/artifacts) 

Is it possible to avoid this? Not a big issue, just asking :wink:

**Comment by lorenzo** [comment]
@zemanlx it is likely not an issue once the final release applies `upx` on it, but I don't think we can reduce the size of the binary without compressing it with upx first

**Comment by lorenzo** [comment]
I'll merge and tag a new release

**Pull request merged**